tests/document/test_instance.py

```diff
@@ -3911,6 +3911,13 @@ def test_dbfields_are_loaded_to_the_right_modelfield_for_dyn_doc_2(self):
             reloaded.z2,
         ) == (doc.x1, doc.x2, doc.y1, doc.y2, doc.z1, doc.z2)
 
+    def test_from_son_with_compiled_regex(self):
+        class User(DynamicDocument):
+            re_str = StringField(regex=r"(^ABC\d\d\d\d$)")
+
+        data = {"re_str": "ABC0000"}
+        User._from_son(son=data, _auto_dereference=False)
```

Comment:
I think it would be better to push the test further and do additional assertions on the returned User instance, for instance check that `new_user.regex is not User.regex`.

Reply:
Since compiled regex objects are atomic, I'll only check regex pattern is same string and StringField is not same object.